File: problems/2996-smallest-missing-integer-greater-than-sequential-prefix-sum/solution.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        n = len(nums)
        length = 1
        for i in range(1, n):
            if nums[i] == nums[i - 1] + 1:
                length += 1
            else:
                break

        prefix_sum = sum(nums[:length])
        s = set(nums)
        while prefix_sum in s:
            prefix_sum += 1

        return prefix_sum

File: problems/2996-smallest-missing-integer-greater-than-sequential-prefix-sum/solution.java
class Solution {
    public int missingInteger(int[] nums) {
        int n = nums.length;
        int longestPrefixLength = 1;
        for (int i = 1; i < n; i++) {
            if (nums[i] == nums[i - 1] + 1) {
                longestPrefixLength++;
            } else {
                break;
            }
        }

        int prefixSum = 0;
        for (int i = 0; i < longestPrefixLength; i++) {
            prefixSum += nums[i];
        }

        boolean[] present = new boolean[101];
        for (int num : nums) {
            present[num] = true;
        }

        int x = prefixSum;
        while (x <= 100 && present[x]) {
            x++;
        }

        while (true) {
            if (!present[x]) {
                return x;
            }
            x++;
        }
    }
}